### DISCUSSION TITLE
Add diagonal watermark with transparency

### DISCUSSION THREAD

**Discussion by skhaz** [ImageMagick/ImageMagick#4254] [Help] [answered]
How can I achieve this watermark using ImageMagick?

![image](https://user-images.githubusercontent.com/46259/134010269-25819cb9-8853-479e-84e9-b974ea6fc45d.jpg)

**Comment by GeeMack** [comment, marked as answer]
With ImageMagick v6 in bash...

    convert image.jpg -alpha set -background none -fill "#ffffff80" \
       \( -size 100x100  label:"PROOF" -rotate 45 -write mpr:tile +delete \) \
       \( +clone -tile mpr:tile -draw "color 0,0 reset" \) -composite result.png

For IMv7 use "magick" instead of "convert". For Windows change the continued-line backslashes "\\" to carets "^", and remove the backslashes "\\(...\\)" that escape the parentheses "(...)".

**Reply by xiebruce** [reply]
what does `mpr` short for?

**Reply by fmw42** [reply]
See https://legacy.imagemagick.org/Usage/files/#mpr

**Reply by xiebruce** [reply]
> See https://legacy.imagemagick.org/Usage/files/#mpr

Thank you. BTW, how can I rotate an image watermark(overlap an image to the base image as a watermark), see [here](https://stackoverflow.com/questions/71599820/imagemagick-add-watermark-proportionally).

**Reply by fmw42** [reply]
Did you read Mark's answer and his comment that you need to set the background to none.

```
magick nature.jpg -set option:logowidth "%[fx:int(w*0.25)]" \
\( logo.png -resize "%[logowidth]x" -background none -rotate 45 \) \
-gravity southeast -geometry +10+10 -composite result.jpg

```
![result](https://user-images.githubusercontent.com/13458218/160254855-f2484735-0aa3-48e5-a278-6dad7070b62b.jpg)

**Reply by xiebruce** [reply]
Yes, I've read Mark's answer, but he didn't mention this. Thank you for answering, it turns out that I should set the background to none, ImageMagick is a little complicated, I still have much to learn.

**Reply by fmw42** [reply]
There is a comment below Mark's answer that says to set the background to none.

**Reply by xiebruce** [reply]
Sorry, maybe I miss that.  

What does `-draw "color 0,0 reset"` exactly mean? Especially the `0,0`, I didn't see any example from [here](https://legacy.imagemagick.org/script/command-line-options.php?#draw)(see the screenshot below)
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/13575500/160278882-b9fb2e08-1151-4e71-ac2b-2b540485cac5.png">

**Reply by xiebruce** [reply]
### Another question

I prefer to set the width and height of the watermark according to the width and height of the image rather than hardcoding it with fixed number `-size 100x100`.

I tried the following command, but as you can see, it throw an error
```bash
convert nature.jpg -set option:logowidth "%[fx:int(w*0.25)]" \
    -set option:logoheight "%[fx:int(h*0.25)]" \
    -alpha set -background none -fill "#ffffff" \
   \( -size "%[logowidth]x%[logoheight]"  label:"THIS IS WATERMARK" -rotate 45 -write mpr:abc +delete \) \
   \( +clone -tile mpr:abc -draw "color 0,0 reset" \) -composite result.png
convert: invalid argument for option '-size': %[logowidth]x%[logoheight] @ error/convert.c/ConvertImageCommand/2852.
```

**Reply by fmw42** [reply]
You need to be using ImageMagick 7 and replace convert with magick for those in-line computations to be accepted as variables.

**Reply by xiebruce** [reply]
Actually I'm using ImageMagick 7, it works after I replace `convert` with `magick`, thank you!
```bash
magick nature.jpg -set option:logowidth "%[fx:int(w*0.25)]" \
-alpha set -background none -fill "#ffffff" \
\( -size "%[logowidth]x"  label:"THIS IS WATERMARK" -rotate 45 -write mpr:abc +delete \) \
\( +clone -tile mpr:abc -draw "color 0,0 reset" \) -composite result.png
```


What about this question?

> What does `-draw "color 0,0 reset"` exactly mean? Especially the `0,0`, I didn't see any example from [here](https://legacy.imagemagick.org/script/command-line-options.php?#draw)(see the screenshot below) <img alt="image" width="1440" src="https://user-images.githubusercontent.com/13575500/160278882-b9fb2e08-1151-4e71-ac2b-2b540485cac5.png">

**Reply by xiebruce** [reply]
Actually I'm using ImageMagick 7, it works after I replace `convert` with `magick`, thank you!
```bash
magick nature.jpg -set option:logowidth "%[fx:int(w*0.25)]" \
-alpha set -background none -fill "#ffffff" \
\( -size "%[logowidth]x"  label:"THIS IS WATERMARK" -rotate 45 -write mpr:abc +delete \) \
\( +clone -tile mpr:abc -draw "color 0,0 reset" \) -composite result.png
```


What about this question?

> What does `-draw "color 0,0 reset"` exactly mean? Especially the `0,0`, I didn't see any example from [here](https://legacy.imagemagick.org/script/command-line-options.php?#draw)(see the screenshot below) <img alt="image" width="1440" src="https://user-images.githubusercontent.com/13575500/160278882-b9fb2e08-1151-4e71-ac2b-2b540485cac5.png">

**Reply by kathiedenissepalafox** [reply]
Increible

**Comment by fmw42** [comment]
See https://legacy.imagemagick.org/Usage/draw/#color

**Reply by xiebruce** [reply]
Got it, thank you.

**Comment by fmw42** [comment]
For new users, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli
https://legacy.imagemagick.org/discourse-server/index.php

**Reply by xiebruce** [reply]
According to [http://www.imagemagick.org/script/command-line-processing.php](http://www.imagemagick.org/script/command-line-processing.php), this command works for me
```bash
convert *.jpeg -resize 50% result_%02d.jpg
```

But if the command is a little complicated, it only processing the first image, then it ends
```bash
convert *.jpg -alpha set -background none -fill "#ffffff80" \
   \( -size 100x100  label:"PROOF" -rotate 45 -write mpr:tile +delete \) \
   \( +clone -tile mpr:tile -draw "color 0,0 reset" \) -composite result_%02d.jpg
```

Should I use bash command `for` loop?

**Comment by snibgo** [comment]
A command that starts...
```
convert *.jpg ...
```
... reads *all* the JPG files. If you need to do the *same* processing on all the images, that works well. But "`+clone`" will make a copy of just the last image in the list for tiling, and "`-composite`" won't do what you want.

Yes, I suggest taking your IM command and putting that inside a shell loop, so the IM command is executed once per input file.
